src/test/RippleKey_test.cpp

```diff
@@ -0,0 +1,287 @@
+            {
+                BEAST_EXPECT(e.what() ==
+                    "Cannot open key file: " + keyFile.string());
+            }
+            remove_all(badPath);
+        }
+
+        key.writeToFile(keyFile);
+
+        auto const key2 = RippleKey::make_RippleKey(keyFile);
+        BEAST_EXPECT(key.keyType() == key2.keyType());
+        BEAST_EXPECT(key.publicKey() == key2.publicKey());
+
+        // Read the keyfile as a Json object to ensure it wrote
+        // what we expected
+        auto const jKeys = [&]
+        {
+            std::ifstream ifsKeys(keyFile.c_str(), std::ios::in);
+
+            if (BEAST_EXPECT(ifsKeys))
+            {
+
+                Json::Reader reader;
+                Json::Value jKeys;
+                BEAST_EXPECT(reader.parse(ifsKeys, jKeys));
+                return jKeys;
+            }
+            return Json::Value{};
+        }();
+
+        using namespace ripple;
+        auto const seed = generateSeed(passphrase);
+        auto const secretKey = generateKeyPair(kt, seed).second;
+
+        // Make sure there are no extra fields
+        BEAST_EXPECT(jKeys.size() == 9);
+        BEAST_EXPECT(jKeys[jss::account_id] ==
+            toBase58(calcAccountID(key.publicKey())));
+        BEAST_EXPECT(jKeys[jss::key_type] == to_string(kt));
+        BEAST_EXPECT(jKeys[jss::master_key] == seedAs1751(seed));
+        BEAST_EXPECT(jKeys[jss::master_seed] == toBase58(seed));
+        BEAST_EXPECT(jKeys[jss::master_seed_hex] ==
+            strHex(seed.data(), seed.size()));
+        BEAST_EXPECT(jKeys[jss::public_key] ==
+            toBase58(TOKEN_ACCOUNT_PUBLIC, key.publicKey()));
+        BEAST_EXPECT(jKeys[jss::public_key_hex] ==
+            strHex(key.publicKey().data(), key.publicKey().size()));
+        BEAST_EXPECT(jKeys["secret_key"] ==
+            toBase58(TOKEN_ACCOUNT_SECRET, secretKey));
+        BEAST_EXPECT(jKeys["secret_key_hex"] ==
+            strHex(secretKey.data(), secretKey.size()));
+    }
+
+    void
+    testSign(ripple::KeyType const kt)
+    {
+        using namespace serialize;
+        using namespace ripple;
+
+        auto const key = RippleKey::make_RippleKey(kt, passphrase);
+
+        auto obj = deserialize(getKnownTx().SerializedText);
+
+        BEAST_EXPECT(obj);
+        ripple::STTx tx{ std::move(*obj) };
+        // The hard-coded version is signed
+        auto check = tx.checkSign(true);
+        BEAST_EXPECT(check.first);
+
+        // Remove the signature
+        auto const origSignature = tx.getFieldVL(sfTxnSignature);
+        auto const origSigningKey = tx.getFieldVL(sfSigningPubKey);
+        tx.makeFieldAbsent(sfTxnSignature);
+        check = tx.checkSign(true);
+        BEAST_EXPECT(!check.first);
+        BEAST_EXPECT(check.second == "Invalid signature.");
+
+        // Now sign it with the test key
+        key.singleSign(tx);
+        BEAST_EXPECT(tx.checkSign(true).first);
+        // Different signature
```

Comment:
The "wrong seed" could be a regular key. The original intent of the test was just to verify that the signature would be validated by `tx.checkSign()`, but I can update to have known signatures.